test/e2e/util_test.go

```diff
@@ -882,3 +884,110 @@ func toggleFeatureGates(deployment *appsv1.Deployment, toToggle ...featuregate.F
+//		Group:   GroupName,
+//		Version: GroupVersion,
+//		Kind:    InstallPlanKind,
+// })
+// Eventually(Apply(plan, func(p *v1alpha1.InstallPlan) error {
+//		p.Spec.Approved = true
+//		return nil
+// })).Should(Succeed())
+func Apply(obj Object, changeFunc interface{}) func() error {
+	// Ensure given object is a pointer
+	objType := reflect.TypeOf(obj)
+	if objType.Kind() != reflect.Ptr {
+		panic(fmt.Sprintf("argument object must be a pointer"))
+	}
+
+	// Ensure given function matches expected signature
+	var (
+		change     = reflect.ValueOf(changeFunc)
+		changeType = change.Type()
+	)
+	if n := changeType.NumIn(); n != 1 {
+		panic(fmt.Sprintf("unexpected number of formal parameters in change function signature: expected 1, present %d", n))
+	}
+	if pt := changeType.In(0); pt != objType {
+		panic(fmt.Sprintf("argument object type does not match the change function parameter type: argument %s, parameter: %s", objType, pt))
+	}
+	if n := changeType.NumOut(); n != 1 {
+		panic(fmt.Sprintf("unexpected number of return values in change function signature: expected 1, present %d", n))
+	}
+	var err error
+	if rt := changeType.Out(0); !rt.Implements(reflect.TypeOf((*error)(nil)).Elem()) {
+		panic(fmt.Sprintf("unexpected return type in change function signature: expected %t, preset %s", err, rt))
+	}
+
+	var (
+		bg     = context.Background()
+		client = ctx.Ctx().Client()
+	)
+	key, err := controllerclient.ObjectKeyFromObject(obj)
+	if err != nil {
+		panic(fmt.Sprintf("unable to extract key from resource: %s", err))
+	}
+
+	return func() error {
+		cp := obj.DeepCopyObject().(Object)
+		if err := client.Get(bg, key, cp); err != nil {
+			return err
+		}
+		cp.GetObjectKind().SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		cp.SetManagedFields(nil)
+
+		var (
+			cpv = reflect.ValueOf(cp)
+			out = change.Call([]reflect.Value{reflect.ValueOf(cp)})
+		)
+		if len(out) != 1 {
+			panic(fmt.Sprintf("unexpected number of return values from apply mutation func: expected 1, returned %d", len(out)))
+		}
+
+		if err := out[0].Interface(); err != nil {
+			return err.(error)
+		}
+
+		if err := client.Patch(bg, cp, controllerclient.Apply, controllerclient.ForceOwnership, controllerclient.FieldOwner("test")); err != nil {
+			return err
+		}
+
+		if err := client.Get(bg, key, cp); err != nil {
+			return err
+		}
+		cp.GetObjectKind().SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		cp.SetManagedFields(nil)
+
+		if len(out) != 1 {
+			panic(fmt.Sprintf("unexpected number of return values from apply mutation func: expected 1, returned %d", len(out)))
+		}
+
+		out = change.Call([]reflect.Value{reflect.ValueOf(cp)})
+		if err := out[0].Interface(); err != nil {
+			return err.(error)
+		}
+
+		if err := client.Status().Patch(bg, cp, controllerclient.Apply, controllerclient.ForceOwnership, controllerclient.FieldOwner("test")); err != nil {
+			return err
+		}
+
+		reflect.ValueOf(obj).Elem().Set(cpv.Elem())
```

Comment:
Does the doc comment make it clear that the value addressed by`obj` will be updated?